service/tablet_allocator.cc

```diff
@@ -1285,6 +1285,21 @@ class load_balancer {
                 maybe_apply({table_plan.current_tablet_count, "current count"});
             }
 
+            utils::get_local_injector().inject("merge_for_missing_data", [&] () {
+                if (s->cf_name() == "test") {
+                    const size_t allow_merge_for_tablet_cnt = 32;
+                    if (table_plan.current_tablet_count == allow_merge_for_tablet_cnt) {
```

Comment:
Why not trigger merge in test by altering keyspace's initial tablet count to 16?

This test-specific code in general logic is not nice.